tests/UnitTests/ValidatableExtensions/ValidatableExtensions.StringPropertiesTests.cs

```diff
@@ -710,4 +712,144 @@ public void ThrowIfPropertyNotContains_WhenPropertyContainsUsingCustomComparison
         // Assert
         action.Should().NotThrow();
     }
+
+    [DataTestMethod]
+    [DataRow("Amichai", @"[a-zA-Z]", RegexOptions.None)]
+    [DataRow("Haman", @"\b[H]\w+", RegexOptions.None)]
+    [DataRow("My Name", @"\bname\b", RegexOptions.IgnoreCase)]
+    public void ThrowIfPropertyMatches_WhenPropertyMatchesRegexPattern_ShouldThrow(string value, string regexPattern, RegexOptions regexOptions)
+    {
+        // Arrange
+        var person = new { Name = value };
+
+        // Act
+        Action action = () => person.Throw().IfMatches(p => p.Name, regexPattern, regexOptions);
+
+        // Assert
+        action.Should()
+            .ThrowExactly<ArgumentException>()
+            .WithMessage($"String should not match RegEx pattern '{regexPattern}' (Parameter '{nameof(person)}: p => p.Name')");
+    }
+
+    [DataTestMethod]
+    [DataRow("123456789", @"[a-zA-Z]", RegexOptions.None)]
+    [DataRow("Amichai", @"\b[H]\w+", RegexOptions.None)]
+    [DataRow("No Match", @"\bname\b", RegexOptions.IgnoreCase)]
+    public void ThrowIfPropertyMatches_WhenPropertyMatchesRegexPattern_ShouldNotThrow(string value, string regexPattern, RegexOptions regexOptions)
```

Comment:
Guessing you meant 👇🏼
```suggestion
    public void ThrowIfPropertyMatches_WhenPropertyNotMatchesRegexPattern_ShouldNotThrow(string value, string regexPattern, RegexOptions regexOptions)
```